modules/social_features/social_event/modules/social_event_managers/src/Form/SocialEventManagersAddEnrolleeForm.php

```diff
@@ -292,6 +309,29 @@ public function buildForm(array $form, FormStateInterface $form_state) {
 
     $form['#cache']['contexts'][] = 'user';
 
+    // We should prevent add enrollments directly if social_event_max_enroll is
+    // enabled and there are no left spots.
+    if ($this->moduleHandler->moduleExists('social_event_max_enroll')) {
```

Comment:
This chunk of code looks identical as in "EnrollInviteEmailForm.php", if this is the case I would recommend to decouple it for example with a trait.

Reply:
Nice suggestion. But then this feature will only land in next major version, as we will be changing modals.